docs/en/reference/query-builder.rst

```diff
@@ -255,6 +255,18 @@ and for managed entities. If you want to set a type explicitly you can call
 the third argument to ``setParameter()`` explicitly. It accepts either a PDO
 type or a DBAL Type name for conversion.
 
+.. note::
+
+    Even though passing DateTime instance is allowed, it impacts performance as by default there is an attempt to load metadata for object, and if it's not found, type is infered from the original value.
+    
+.. code-block:: php
+
+    <?php
+    
+    use Doctrine\DBAL\Types\Types;
+    
+    $qb->setParameter('date', new \DateTimeImmutable(), Types::DATE_IMMUTABLE) // prevents attempt to load metadata for date time class, improving performance
```

Comment:
This is going to translate into a horizontal scroll, and your comment, which is the most important thing to read, will be hidden by default
I'd either do this (a bit weird but it's more obvious that we are talking about the 3rd element here):

```suggestion
    $qb->setParameter(
        'date',
        new \DateTimeImmutable(),
        Types::DATE_IMMUTABLE /* prevents attempt to load metadata 
        for date time class, improving performance */
    )
```

or that (a bit easier on the eye):

```suggestion
     // prevents attempt to load metadata for date time class, improving performance
    $qb->setParameter('date', new \DateTimeImmutable(), Types::DATE_IMMUTABLE)
```